modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/KeyValueProcessor.java

```diff
@@ -105,7 +111,9 @@ public void execute(IngestDocument document) {
                 }
                 return kv;
             })
-            .filter((p) -> includeKeys == null || includeKeys.contains(p[0]))
+            .filter((p) ->
+                (includeKeys == null || includeKeys.contains(p[0])) &&
+                    (excludeKeys == null || excludeKeys.contains(p[0]) == false))
```

Comment:
thanks, I suppose I assumed the cost is minimal since this list is not expected to be more than just a few items... but that is being multiplied by millions of documents, so good call